### ISSUE TITLE
Itemize the types of search that CALC supports

### ISSUE THREAD

**Issue by ericronne** [18F/calc#2039]
To inform interface design …

**Comment by ericronne** [comment]
This page from the api documentation _might_ help a little …
https://calc.gsa.gov/api/docs/

**Comment by ericronne** [comment]
[Google doc](https://docs.google.com/document/d/1wNCCsNkccDZVmGXXqDHu-zkjhW33MMRa2pBE9J2N6Ro/edit?usp=sharing) to hash out some content.

### Who's using the radio buttons? Not many, it seems …
![image](https://user-images.githubusercontent.com/9259185/42527153-7f859a5a-843d-11e8-92aa-a9a7a0d9f81f.png)

**Comment by tbaxter-18f** [comment]
The search, as currently implemented:

1. Cleans the search query string and searches for matching labor categories.
  - Exact match will match the input string exactly. If it's a comma-separated list, each comma-separated item is matched independently, and each must be an exact match..
  - Otherwise any quoted terms are matched exactly input are matched as input, but it can be a partial match. So, for example, "accounting clerk" will find "Accounting Clerk II". Unquoted terms are broken apart into individual words and will match any labor category in which all of the words in the term are found.
2. After items matching the query string are found, any exclusions are removed from those results
3. Results are filtered for experience (if submitted)
4. Results are filtered for education level (if submitted)
5. Results are filtered for schedule (if submitted)
6. Results are filtered for site (if submitted)
7. Results are filtered for SIN and price (if submitted -- but note that there do not appear to be fields on the page to submit these values)
8. Results are sorted 

Does that answer the question and resolve this?

**Comment by ericronne** [comment]
Thanks! Could you summarize how CALC handles terms which _aren't_ in quotes? (I'm sure you could do so more clearly than i could :) )

**Comment by tbaxter-18f** [comment]
Sure, see up above where it says "Unquoted terms are broken apart into individual words and will match any labor category in which all of the words in the term are found"? What that means in practice is that for any search with "exact match" unchecked:

1. if the search is `business analyst` it will look for any labor category with `business` AND `analyst` in it.
2. if the search is `business analyst, program analyst` it will get results with `business` AND `analyst` in the labor category OR `program` AND `analyst`
3. if the search is `business analyst, "program analyst"` it will get results with `business` AND `analyst` in the labor category OR `program analyst`. The difference from number two and number three is that in scenario two `program sr analyst` would match, and in scenario three it would not. However `senior program analyst` would match either way. 

In practice, I suspect few people use quotes around phrases, but if they do we have them covered.

**Comment by ericronne** [comment]
Crystal 💎

I think this will serve as a good reference as we iterate on the search utility, thanks again ❤️
